tests/unit/s2n_x509_validator_test.c

```diff
@@ -1262,6 +1386,115 @@ int main(int argc, char **argv)
         s2n_x509_validator_wipe(&validator);
         s2n_x509_trust_store_wipe(&trust_store);
     };
+
+    /**
+     * Test OCSP validation at various offsets from update times.
+     *
+     * libcrypto ASN1 comparison calculates differences in terms of days and seconds,
+     * so try mocking the system time to a collection of more than day & less
+     * than day differences.
+     * The T's in the below diagram represent test cases that should fail
+     * The F's represent test cases that should succeed
+     *    S2N_ERR_CERT_INVALID                            S2N_ERR_CERT_EXPIRED
+     *          |   |                                        |   |
+     *          v   v                                        v   v
+     *          F   F   T   T                        T   T   F   F
+     *          v   v   v   v                        v   v   v   v
+     * <----------|---|---|----------------------------|---|---|--->
+     *                ^                                    ^
+     *       this update                                 next update
+     *                |---|
+     *                  one day
+     *
+     * If this test is failing make sure that the this_update_timestamp_nanoseconds
+     * matches the actual timestamp of ocsp_response_early_expire.der
+     *
+     * openssl ocsp -respin ocsp_response_early_expire.der -text -noverify | grep "This Update"
+     */
+    {
+        /* Apr 28 22:11:56 2023 GMT */
+        uint64_t this_update_timestamp_nanoseconds = (uint64_t) 1682719916 * ONE_SEC_IN_NANOS;
+
+        /* Apr 28 22:11:56 2023 GMT */
+        uint64_t next_update_timestamp_nanoseconds = (uint64_t) 2082838316 * ONE_SEC_IN_NANOS;
+
+        uint64_t one_hour_nanoseconds = (uint64_t) 60 * 60 * ONE_SEC_IN_NANOS;
+        uint64_t one_day_nanoseconds = 24 * one_hour_nanoseconds;
+
+        uint64_t times[] = {
+            this_update_timestamp_nanoseconds - (one_day_nanoseconds + one_hour_nanoseconds),
+            this_update_timestamp_nanoseconds - one_hour_nanoseconds,
+            this_update_timestamp_nanoseconds + one_hour_nanoseconds,
+            this_update_timestamp_nanoseconds + (one_day_nanoseconds + one_hour_nanoseconds),
+            next_update_timestamp_nanoseconds - (one_day_nanoseconds + one_hour_nanoseconds),
+            next_update_timestamp_nanoseconds - one_hour_nanoseconds,
+            next_update_timestamp_nanoseconds + one_hour_nanoseconds,
+            next_update_timestamp_nanoseconds + (one_day_nanoseconds + one_hour_nanoseconds)
+        };
+
+        int expected_result[] = {
```

Comment:
The cleaner way to do this might be:
```
    struct {
        int result;
        uint64_t time;
    } test_cases[] = {
        {  .result = S2N_ERR_CERT_INVALID, .time = this_update_timestamp_nanoseconds - (one_day_nanoseconds + one_hour_nanoseconds)  },
        ....
    };

    for (size_t i = 0; i < s2n_array_len(test_cases); i++) {
        printf("%i %i\n", test_cases[i].result, test_cases[i].time);
    }
```
Otherwise, the relationship between your times and your results is harder to follow and easier to mess up.